Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -836,19 +836,26 @@ public final class PackageBuilder {
     }
 
     /// Returns the list of platforms supported by the manifest.
-    func platforms() -> [SupportedPlatform] {
-        if let platforms = _platforms {
+    func platforms(isTest: Bool = false) -> [SupportedPlatform] {
+        if let platforms = _platforms[isTest] {
             return platforms
         }
 
         var supportedPlatforms: [SupportedPlatform] = []
 
         /// Add each declared platform to the supported platforms list.
         for platform in manifest.platforms {
+            let declaredPlatform = platformRegistry.platformByName[platform.platformName]!
+            var version = PlatformVersion(platform.version)
+
+            // macOS tests have a higher minimum deployment target
+            if declaredPlatform == Platform.macOS && isTest && version < PlatformVersion("10.15") {
+                version = PlatformVersion("10.15")
```

Comment:
Can you declare this as a constant in order to avoid potential typos:

```swift
extension PlatformVersion {
    static let v10_15: PlatformVersion = PlatformVersion("10.15")
}
```